src/main/java/org/breedinginsight/brapi/v2/services/BrAPIGermplasmService.java

```diff
@@ -0,0 +1,92 @@
+import org.breedinginsight.brapps.importer.model.ImportUpload;
+import org.breedinginsight.brapps.importer.model.base.Germplasm;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.*;
+
+@Slf4j
+@Singleton
+public class BrAPIGermplasmService {
+
+    private BrAPIGermplasmDAO germplasmDAO;
+    private final String BREEDING_METHOD_ID_KEY = "breedingMethodId";
+
+    // TODO: Test accession number match in germplasm (tests might cover this)
+    // TODO: Test germplasm importer
+
+    @Inject
+    public BrAPIGermplasmService(BrAPIGermplasmDAO germplasmDAO) {
+        this.germplasmDAO = germplasmDAO;
+    }
+
+    public List<BrAPIGermplasm> getGermplasm(UUID programId) {
+        // Get germplasm
+        List<BrAPIGermplasm> germplasmList;
+        try {
+            germplasmList = germplasmDAO.getGermplasm(programId);
+        } catch (ApiException e) {
+            throw new InternalServerException(e.getMessage());
+        }
+
+        // Process the germplasm
+        Map<String, BrAPIGermplasm> germplasmByFullName = new HashMap<>();
+        for (BrAPIGermplasm germplasm: germplasmList) {
+            germplasmByFullName.put(germplasm.getGermplasmName(), germplasm);
+
+            JsonObject additionalInfo = germplasm.getAdditionalInfo();
+            if (additionalInfo != null && additionalInfo.has(BREEDING_METHOD_ID_KEY)) {
+                germplasm.setBreedingMethodDbId(additionalInfo.get(BREEDING_METHOD_ID_KEY).getAsString());
+            }
+
+            if (germplasm.getDefaultDisplayName() != null) {
+                germplasm.setGermplasmName(germplasm.getDefaultDisplayName());
+            }
+        }
+
+        // Update pedigree string
+        for (BrAPIGermplasm germplasm: germplasmList) {
+            if (germplasm.getPedigree() != null) {
+                String newPedigreeString = germplasm.getPedigree();
+                List<String> parents = Arrays.asList(germplasm.getPedigree().split("/"));
+                if (parents.size() >= 1 && germplasmByFullName.containsKey(parents.get(0))) {
+                    newPedigreeString = germplasmByFullName.get(parents.get(0)).getAccessionNumber();
+                }
+                if (parents.size() == 2 && germplasmByFullName.containsKey(parents.get(1))) {
+                    newPedigreeString += "/" + germplasmByFullName.get(parents.get(1)).getAccessionNumber();
+                }
+                germplasm.setPedigree(newPedigreeString);
+            }
+        }
+
+        return germplasmList;
+    }
+
+    public List<BrAPIGermplasm> importBrAPIGermplasm(List<BrAPIGermplasm> brAPIGermplasmList, UUID programId, ImportUpload upload) throws ApiException {
+        return germplasmDAO.importBrAPIGermplasm(brAPIGermplasmList, programId, upload);
+    }
+
+    public List<BrAPIGermplasm> getGermplasmByAccessionNumber(ArrayList<String> germplasmAccessionNumbers, UUID id) throws ApiException {
```

Comment:
Can the `id` paremeter name be changed to `programId`?

Reply:
changed